### ISSUE TITLE
renderDigitsOrdinal looks off in some languages

### ISSUE THREAD

**Issue by crisbeto** [papandreou/node-cldr#33]
The generated `renderDigitsOrdinal` number function seems off. Example:

```js
console.log(cldr.extractRbnfFunctionByType('en-us').renderDigitsOrdinal(1));
// logs "1ordinalonesttwondfewrdotherth"
```

Looking at what gets generated, it looks like this:

```js
function anonymous(n) {
    if (n < 0) return "−" + this.renderDigitsOrdinal(-n);
    if (n >= 0) return this.renderNumber(n, "#,##0") + "ordinal" + "one" + "st" + "two" + "nd" + "few" + "rd" + "other" + "th"
}
```

The function clearly isn't meant to return the formatted number with `ordinalonesttwondfewrdotherth`. I got here after looking at https://github.com/rikkertkoppes/rbnf which uses v2.0.1 that doesn't have the issue, so my guess is that it got broken somewhere between 2.0 and 3.5 (which is the version I'm using).

**Comment by crisbeto** [comment]
It seems to be due to the `en.xml` getting changed at some point:

```xml
<!-- new -->
<rulesetGrouping type="OrdinalRules">
  <ruleset type="digits-ordinal">
      <rbnfrule value="-x">−→→;</rbnfrule>
      <rbnfrule value="0">=#,##0=$(ordinal,one{st}two{nd}few{rd}other{th})$;</rbnfrule>
  </ruleset>
</rulesetGrouping>

<!-- old -->
<rulesetGrouping type="OrdinalRules">
    <ruleset type="digits-ordinal-indicator" access="private">
        <rbnfrule value="0">th;</rbnfrule>
        <rbnfrule value="1">st;</rbnfrule>
        <rbnfrule value="2">nd;</rbnfrule>
        <rbnfrule value="3">rd;</rbnfrule>
        <rbnfrule value="4">th;</rbnfrule>
        <rbnfrule value="20">→→;</rbnfrule>
        <rbnfrule value="100">→→;</rbnfrule>
    </ruleset>
    <ruleset type="digits-ordinal">
        <rbnfrule value="-x">−→→;</rbnfrule>
        <rbnfrule value="0">=#,##0==%%digits-ordinal-indicator=;</rbnfrule>
    </ruleset>
</rulesetGrouping>
```

**Comment by papandreou** [comment]
Thanks for discovering this problem and looking into it. Seems like the format change/addition is mentioned here: http://cldr.unicode.org/index/downloads/cldr-26#TOC-RBNF

**Comment by papandreou** [comment]
Fixed in 4.2.0.
